>From 7f782a324333fce427d2369b7ec96ae99f61f8c5 Mon Sep 17 00:00:00 2001
From: Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
Date: Thu, 10 Mar 2016 12:11:54 +0100
Subject: [PATCH] Improve detection of shell scripts

Affects [checkbashisms] and [shellcheck]
---
 data/sh | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/data/sh b/data/sh
index 2389117..32469e6 100644
--- a/data/sh
+++ b/data/sh
@@ -4,13 +4,23 @@ command = sh -n {file}
 
 [checkbashisms]
 apt = devscripts
-files = *.sh
-command = checkbashisms {files}
+# TODO: replace not-dirs with --ignore option
+not-dirs = .git .svn .bzr CVS .hg _darcs _FOSSIL_ .sgdrawer
+# TODO: replace not-files with recursive option (#780197)
+not-files =
+ configure config.status config.sub config.guess install-sh install.sh
+# Careful: file -F "<LITERAL_TAB>" ... cut -d "<LITERAL_TAB>"
+command = file -F "	" -N {files} | grep -a 'shell script' | cut -f 1 -d "	" | xargs -d"\n" --no-run-if-empty checkbashisms
 
 [shellcheck]
 apt = shellcheck
-files = *.sh *.bash *.zsh
-command = shellcheck {files}
+# TODO: replace not-dirs with --ignore option
+not-dirs = .git .svn .bzr CVS .hg _darcs _FOSSIL_ .sgdrawer
+# TODO: replace not-files with recursive option (#780197)
+not-files =
+ configure config.status config.sub config.guess install-sh install.sh
+# Careful: file -F "<LITERAL_TAB>" ... cut -d "<LITERAL_TAB>"
+command = file -F "	" -N {files} | grep -a 'shell script' | cut -f 1 -d "	" | xargs -d"\n" --no-run-if-empty shellcheck
 
 [bashate]
 apt = python3-bashate | python-bashate
-- 
2.7.0

